django/urls/converters.py

```diff
@@ -0,0 +1,73 @@
+import uuid
+
+from django.utils import lru_cache
+
+
+class IntConverter:
+    regex = '[0-9]+'
+
+    def to_python(self, value):
+        return int(value)
+
+    def to_url(self, value):
+        return str(value)
+
+
+class StringConverter:
+    regex = '[^/]+'
+
+    def to_python(self, value):
+        return value
+
+    def to_url(self, value):
+        return value
+
+
+class UUIDConverter:
+    regex = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
+
+    def to_python(self, value):
+        return uuid.UUID(value)
+
+    def to_url(self, value):
+        return str(value)
+
+
+class SlugConverter(StringConverter):
+    # Django has both `slug_re` and `slug_unicode_re`. In the case of URLs, the
+    # non-unicode variant seems to make more sense.
+    regex = '[-a-zA-Z0-9_]+'
+
+
+class PathConverter(StringConverter):
+    regex = '.+'
+
+
+DEFAULT_CONVERTERS = {
+    'int': IntConverter(),
+    'path': PathConverter(),
+    'slug': SlugConverter(),
+    'str': StringConverter(),
+    'uuid': UUIDConverter(),
+}
+
+
+REGISTERED_CONVERTERS = {}
+
+
+def register_converter(converter, typename):
```

Comment:
I guess `typename` is a thing in C++ but I think `type_name` is more Pythonic. Any objection?

Reply:
None at all. It's also not specified to that extend in the DEP.